opencti-platform/opencti-graphql/src/domain/settings.js

```diff
@@ -197,3 +199,19 @@ export const getCriticalAlerts = async (context, user) => {
   // no alert
   return [];
 };
+
+// Retrieves max level of markings that can be shared.
+export const getDataSharingMaxMarkings = async (context, user, settings) => {
+  const { platform_data_sharing_max_markings } = settings ?? await getEntityFromCache(context, user, ENTITY_TYPE_SETTINGS);
+  if (!platform_data_sharing_max_markings) {
+    return [];
+  }
+  return internalFindByIds(context, user, platform_data_sharing_max_markings);
```

Comment:
I suggest to use cache instead of querying markings in DB
```suggestion
  const dataSharingMaxMarkings = platform_data_sharing_max_markings ?? [];
  const allMarkingsMap = await getEntitiesMapFromCache(context, SYSTEM_USER, ENTITY_TYPE_MARKING_DEFINITION);
  return dataSharingMaxMarkings.map((m) => allMarkingsMap.get(m) ?? m);
```